nucypher/blockchain/eth/domains.py

```diff
@@ -18,38 +24,69 @@ class PolygonChain(ChainInfo, Enum):
     MUMBAI = (80001, "mumbai")
 
 
-class DomainInfo(NamedTuple):
-    name: str
-    eth_chain: EthChain
-    polygon_chain: PolygonChain
+class TACoDomain:
+    def __init__(self, name: str, eth_chain: EthChain, polygon_chain: PolygonChain):
+        self.name = name
+        self.eth_chain = eth_chain
+        self.polygon_chain = polygon_chain
+
+    def __repr__(self) -> str:
+        return f"<TACoDomain {self.name}>"
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __hash__(self) -> int:
+        return hash(self.name)
+
+    def __bytes__(self) -> bytes:
+        return self.name.encode()
+
+    def __eq__(self, other: Any) -> bool:
+        try:
+            return self.name == other.name
```

Comment:
As long as we're avoiding name collisions what's the benefit of a more complex equality check?

Reply:
I guess the question I have is, if `TACoDomain` is a public class, what is stopping someone from creating their own `TACoDomain` object by doing the following (perhaps by mistake):
```python
myDomain = TACoDomain("mainnet", EthChain.GOERLI, PolygonChain.MUMBAI)
...
```
and passing it around as the domain? If there is indeed something that prevents that, then great!

Reply:
Nothing is preventing that however that seems like acceptable behavior to me...?  (technically any object with a matching name attribute will eval truthy) It's pretty hard misuse by mistake imo, as you've demonstrated you'd need to be interacting with domain construction/mutation to begin with, then set the name incorrectly.  Changing it may have subtle side-effects... I need to think about it a bit

Reply:
✔️  after carefully checking it seems a more complex check has desirable effects!  This is a more correct and rigid design especially since we're able to compare enum instances directly.  Need to think about expanding this check in #3318

Reply:
👍

Reply:
(also updated `__hash__` for consistency)